File: 11.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int maxim = 0, i = 0, n = height.size()-1;
        while(i < n){
            int m = min(height[i], height[n]);
            maxim = max(maxim, m*(n-i));
            if(height[n] > height[i]){
                i++;
            } else{
                n--;
            }
        }
        return maxim;
    }
};

File: 11.go
package main

func maxArea(height []int) int {
	i, j := 0, len(height)-1
	ans := 0
	for i < j {
		ans = Max(ans, (j-i)*Min(height[i], height[j]))
		if height[j] > height[i] {
			i++
		} else {
			j--
		}
	}
	return ans
}

func Min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func Max(a, b int) int {
	if a > b {
		return a
	}
	return b
}
